### PR TITLE
Fix printing of dependencies 

### PR DESCRIPTION
**Pull request by robertzk**

### CODE REVIEW

File: R/align.R [hunk 1]
```diff
@@ -2,6 +2,7 @@
 #'
 #' @param locked_package locked_package. In particular, the \code{version}
 #'    and \code{name} elements will be used.
+#' @param verbose logical. Whether or not to announce usage of the package.
```
Comment (peterhurford) [inline R/align.R]: This param isn't actually included in the function though?

File: R/align.R [hunk 2]
```diff
@@ -17,6 +18,8 @@ align <- function(locked_package) {
 
   ## Symlink the locked package to the correct lockbox version.
   `symlink_to_lockbox!`(locked_package)
+
+  announce_package_usage(locked_package$name, locked_package$version)
```
Comment (peterhurford) [inline R/align.R]: `if (isTRUE(verbose)) { }` ?

File: R/lockbox.R [hunk 3]
```diff
@@ -132,8 +132,12 @@ lockbox_download_dir <- function() {
 
 #' The transient lockbox library path.
 lockbox_transient_dir <- function() {
-  getOption("lockbox.transient_dir") %||%
+  if (!is.null(.lockbox_env$session_id)) {
```
Comment (kirillseva) [inline R/lockbox.R]: 👍

File: R/utils.R [hunk 4]
```diff
@@ -16,6 +16,14 @@ lockbox_imports <- function() {
   Filter(nzchar, strsplit(read.dcf(dcf)[,"Imports"], "[\n, ]+")[[1]])
 }
 
+announce_package_usage <- function(name, version, dependency = FALSE) {
```
Comment (kirillseva) [inline R/utils.R:19]: unless you want to pass verbose argument here set to FALSE by default, and then call this function with verbose = TRUE explicitly if you want things printed

File: R/zzz.R [hunk 5]
```diff
@@ -1,6 +1,10 @@
 .lockbox_env <- new.env(parent = emptyenv())
 
 set_session_id <- function() {
+  if (isTRUE(getOption("lockbox.disable_sessions"))) {
+    return()
+  }
```
Comment (kirillseva) [inline R/zzz.R:6]: can one-line this function, optionally 🦀

File: R/zzz.R [hunk 6]
```diff
@@ -35,6 +38,17 @@ set_transient_library <- function() {
     dir.create(transient_staging_path, FALSE, TRUE)
   }
   .libPaths(c(transient_staging_path, dir, .libPaths()))
+
+  # Finally, we initialize the transient library with some sane default symlinks.
+  with_real_packages(lockbox_library(), function(pkgpath) {
+    versions <- list.files(pkgpath)
```
Comment (kirillseva) [inline R/zzz.R:44]: why not `versions <- as.package_version(list.files(pkgpath))` here? then you can get rid of the comment below

File: R/utils.R (hunk 4 above)
Comment (robertzk) [reply R/utils.R:19]: Printing is all this function does, though.

File: R/zzz.R (hunk 6 above)
Comment (robertzk) [reply R/zzz.R:44]: :man_facepalming: Great point.

File: R/lockbox.R [hunk 7]
```diff
@@ -55,9 +55,9 @@ lockbox.list <- function(lock, env) {
 
   sapply(lock[!load_these_packages], function(locked_package) {
     if (locked_package$is_dependency_package) {
-      cat('Using dependency', crayon_blue(locked_package$name), as.character(locked_package$version), '\n')
+      announce_package_usage(locked_package$name, locked_package$version, TRUE)
```
Comment (peterhurford) [inline R/lockbox.R:58]: Might be nice to specify variable names here so people know what they mean.

File: R/lockbox.R (hunk 3 above)
Comment (russellpierce) [reply R/lockbox.R]: Nitpicking question; isn't this breaking the pattern a little by forcing a reassignment of the option on every function call when .lockbox_env$session_id is not NULL?  Couldn't the infix operator be followed by an ifelse? Maybe there is something else you're trying to accomplish here that I'm not following. :(

File: R/zzz.R (hunk 5 above)
Comment (russellpierce) [reply R/zzz.R:6]: Should return be an explicit NULL for readablity?

Comment (robertzk) [reply R/zzz.R:6]: I've generally used `return()` for functions that don't expect to return values anyway and are only called for their side effects.

File: R/lockbox.R (hunk 3 above)
Comment (robertzk) [reply R/lockbox.R]: @drknexus Good point. I didn't like the complexity of following `%||%` with `ifelse` but I'll re-order this a bit.

File: R/lockbox.R [hunk 8]
```diff
@@ -122,18 +122,29 @@ is.locked_package <- function(obj) { methods::is(obj, "locked_package") }
...
 }
 
 #' The transient lockbox library path.
 lockbox_transient_dir <- function() {
-  getOption("lockbox.transient_dir") %||%
-    normalizePath(file.path("~/.R/lockbox_transient", .lockbox_env$session_id), mustWork = FALSE)
+  if (!is.null(getOption("lockbox.transient_dir"))) {
+    getOption("lockbox.transient_dir")[1L]
+  } else if (!is.null(.lockbox_env$session_id)) {
+    normalizePath(file.path("~/.R/lockbox_transient_sessions", .lockbox_env$session_id), mustWork = FALSE)
```
Comment (robertzk) [inline R/lockbox.R:144]: Chosen to prevent cross-pollution.

**Comment by robertzk** [comment]
@peterhurford @kirillseva Final review?

File: DESCRIPTION [hunk 9]
```diff
@@ -24,4 +24,4 @@ Suggests:
     uuid
 Roxygen: list(wrap = FALSE)
 VignetteBuilder: knitr
-RoxygenNote: 6.0.1
```
Comment (peterhurford) [inline DESCRIPTION]: What happened here?

**Comment by peterhurford** [comment]
Consistency on Roxygen would be appreciated, but looks good otherwise!

**Comment by peterhurford** [comment]
LGTM :+1: :shipit:

**Comment by robertzk** [comment]
Tested with sessions enabled and disabled. Will merge when CI is green.

**Pull request merged**